pages/docs/reference/properties.md

```diff
@@ -179,13 +179,26 @@ public class MyTest {
 }
 ```
 
-The modifier can only be used on `var` properties declared inside the body of a class (not in the primary constructor), and only
-when the property does not have a custom getter or setter. The type of the property must be non-null, and it must not be
-a primitive type.
+The modifier can be used on `var` properties declared inside the body of a class (not in the primary constructor, and only
+when the property does not have a custom getter or setter) and, since Kotlin 1.2, for top-level properties and 
+local variables. The type of the property or variable must be non-null, and it must not be a primitive type.
 
 Accessing a `lateinit` property before it has been initialized throws a special exception that clearly identifies the property
 being accessed and the fact that it hasn't been initialized.
 
+### Checking whether a lateinit var is initialized (since 1.2)
+
+To check whether a `lateinit var` has already been initialized, use `.isInitialized` on 
+the [reference to that property](reflection.html#property-references):
+
+```kotlin
+if (foo::bar.isInitialized) {
+    println(foo.bar)
+}
+```
+
+This check is only available for the properties declared in the same class or one of the outer classes.  
```

Comment:
Not sure how this applies to properties which aren't declared in a class. I guess this needs a bit more clarification.

Reply:
Done.